rust/lance-encoding/src/data.rs

```diff
@@ -233,25 +231,57 @@ impl<T: OffsetSizeTrait> DataBlockBuilderImpl for VariableWidthDataBlockBuilder<
         let block = data_block.as_variable_width_ref().unwrap();
         assert!(block.bits_per_offset == T::get_byte_width() as u8 * 8);
 
-        let offsets: ScalarBuffer<T> = block.offsets.clone().borrow_to_typed_slice();
```

Comment:
`borrow_to_typed_slice` is much heavier than I expected. It’s much faster to just read the offset one by one. Maybe we can extract something like `OffsetView` for this? Inside that view, we’d use it as a slice if it’s aligned, otherwise read it one by one.

What do you think?

Reply:
I don't think `borrow_to_typed_slice` clones the array?  It should only clone if the data is misaligned.

Reply:
You are also correct on this!

After a deeper dive, I’m now confident that the buffer was aligned and the cost comes from heavy Arc creation and destruction. Our fix just happened to work around it, so there’s nothing about misalignment.

I’ll work on a more complete fix for this.

Reply:
I tried another approach for this, it’s much more beautiful.